cmd/bugzilla-backporter/main.go

```diff
@@ -0,0 +1,137 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/openshift/ci-tools/pkg/backporter"
+	"github.com/prometheus/client_golang/prometheus"
+	log "github.com/sirupsen/logrus"
+	"k8s.io/test-infra/pkg/flagutil"
+	prowConfig "k8s.io/test-infra/prow/config"
+	"k8s.io/test-infra/prow/config/secret"
+	prowflagutil "k8s.io/test-infra/prow/flagutil"
+	"k8s.io/test-infra/prow/interrupts"
+	"k8s.io/test-infra/prow/metrics"
+	"k8s.io/test-infra/prow/pjutil"
+)
+
+var (
+	bzbpMetrics = struct {
+		httpRequestDuration *prometheus.HistogramVec
+		httpResponseSize    *prometheus.HistogramVec
+		errorRate           *prometheus.CounterVec
+	}{
+		httpRequestDuration: prometheus.NewHistogramVec(
+			prometheus.HistogramOpts{
+				Name:    "bugzilla_backporter_http_request_duration_seconds",
+				Help:    "http request duration in seconds",
+				Buckets: []float64{0.0005, 0.001, 0.0025, 0.005, 0.01, 0.025, 0.05, 0.1, 0.25, 0.5, 1, 2},
+			},
+			[]string{"status", "path"},
+		),
+		httpResponseSize: prometheus.NewHistogramVec(
+			prometheus.HistogramOpts{
+				Name:    "bugzilla_backporter_http_response_size_bytes",
+				Help:    "http response size in bytes",
+				Buckets: []float64{256, 512, 1024, 2048, 4096, 6144, 8192, 10240, 12288, 16384, 24576, 32768, 40960, 49152, 57344, 65536},
+			},
+			[]string{"status", "path"},
+		),
+		errorRate: prometheus.NewCounterVec(
+			prometheus.CounterOpts{
+				Name: "bugzilla_backporter_error_rate",
+				Help: "number of errors, sorted by label/type",
+			},
+			[]string{"error"},
+		),
+	}
+)
+
+func recordError(label string, path string) {
+	labels := prometheus.Labels{"error": label, "path": path}
+	bzbpMetrics.errorRate.With(labels).Inc()
+}
+
+type options struct {
+	logLevel    string
+	address     string
+	gracePeriod time.Duration
+	bugzilla    prowflagutil.BugzillaOptions
+}
+
+type traceResponseWriter struct {
+	http.ResponseWriter
+	statusCode int
+	size       int
+}
+
+func handleWithMetrics(h backporter.HandlerFuncWithErrorReturn) http.HandlerFunc {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
My IDE says this conversion is unnecessary.

Reply:
Same thing is three times in `backporter_test.go`

Reply:
Yup. Those conversions are redundant, I'll remove them.